runtime/ztests/expr/function/len.yaml

```diff
@@ -0,0 +1,89 @@
+zed: len(a)
+
+vector: true
+
+input: |
+  {a:null}
+  {a:null(bytes)}
+  {a:0x}
+  {a:0x0123}
+  {a:0x0123((bytes,string))}
+  {a:null(string)}
+  {a:""}
+  {a:"string"}
+  {a:null(ip)}
+  {a:1.0.0.0}
+  {a:1::}
+  {a:null(net)}
+  {a:1.0.0.0/8}
+  {a:1::/8}
+  {a:null({})}
+  {a:{}}
+  {a:{x:1}}
+  {a:{x:1,s:"hello"}}
+  {a:{x:{a:1,b:2}}}
+  {a:null([int64])}
+  {a:[]([int64])}
+  {a:[1,2]([int64])}
+  {a:null(|[int64]|)}
+  {a:|[]|(|[int64]|)}
+  {a:|[1,2]|(|[int64]|)}
+  {a:null(|{int64:int64}|)}
+  {a:|{1:1}|(|{int64:int64}|)}
+  {a:|{1:1,2:2}|}
+  {a:<int64>}
+  {a:<type>}
+  {a:<{x:int64,y:int64}>}
+  {a:<(int64,string)>}
+  {a:<[int64]>}
+  {a:<[(int64,string)]>}
+  {a:<any=(int64,string)>}
```

Comment:
Nit: Wrapping these in a record makes them harder to read without any apparent benefit.